CriticalMass/IdentifiableAnnnotation.swift

```diff
@@ -19,13 +19,14 @@ class IdentifiableAnnnotation: MKPointAnnotation {
 
     var location: Location {
         didSet {
-            coordinate = CLLocationCoordinate2D(latitude: location.latitude, longitude: location.longitude)
+            coordinate = CLLocationCoordinate2D(location)
         }
     }
 
     init(location: Location, identifier: String) {
         self.identifier = identifier
         self.location = location
         super.init()
+        coordinate = CLLocationCoordinate2D(location)
```

Comment:
You could set location in a defer statement which would trigger didSet. That way coordinated would get set only from one place afai can see in review mode on my phone.

Reply:
Interesting, didn't know `defer` could help in this case (I found an example here: https://stackoverflow.com/a/33979852/6716961). But this can be only used with Optionals, which we could do but I don't know if it's worth it. 

It would then look like this: 
```swift
var location: Location? {
        didSet {
            guard let location = location else { return }
            coordinate = CLLocationCoordinate2D(location)
        }
    }

    init(location: Location, identifier: String) {
        self.identifier = identifier
        defer {
            self.location = location
        }
        super.init()
    }
```

Reply:
Right. With `Location` as the type for this it would need to be optional. I am wondering though why we pass that type around anyways since it is just the coordinate that is needed. When that would be CLCoordinate2D the initial state could be `kCLLocationCoordinate2DInvalid` and then use defer. 
But I guess that would need a rebuild of the location manager which is sending out the` Location ` type which I think is a flaw. I am working on a rewrite of the manager btw. as a first step to move away from notification based data flow which is often used

Reply:
+++ Breaking +++
Just read that the defer usage in the init is considered a bug by the swift core team 😅 So better not use it